tools/pipeline-witness/infrastructure/Assign-StoragePermissions.ps1

```diff
@@ -0,0 +1,70 @@
+#>
+param(
+  [Parameter(Mandatory)]
+  [validateSet('staging', 'test')]
+  [string]$target
+)
+
+function Invoke([string]$command) {
+  Write-Host "> $command"
+  Invoke-Expression $command
+}
+
+Push-Location $PSScriptRoot
+try {
+  $subscriptionName = $target -eq 'test' ? 'Azure SDK Developer Playground' : 'Azure SDK Engineering System'
+  Write-Host "Setting subscription to '$subscriptionName'"
+  Invoke "az account set --subscription '$subscriptionName' --output none"
+
+  $parametersFile = "./bicep/parameters.$target.json"
+  Write-Host "Reading parameters from $parametersFile"
+  $parameters = (Get-Content -Path $parametersFile -Raw | ConvertFrom-Json).parameters
+  $appResourceGroupName = $parameters.appResourceGroupName.value
+  $appStorageAccountName = $parameters.appStorageAccountName.value
+  $cosmosAccountName = $parameters.cosmosAccountName.value
+  $logsResourceGroupName = $parameters.logsResourceGroupName.value
+  $logsStorageAccountName = $parameters.logsStorageAccountName.value
+
+  Write-Host "Adding Azure SDK Engineering System Team RBAC access to storage resources:`n" + `
+    "  Blob: $logsResourceGroupName/$logsStorageAccountName`n" + `
+    "  Queue: `n" + `
+    "  Cosmos: $appResourceGroupName/$cosmosAccountName`n"
+
+  Write-Host "Getting group id for Azure SDK Engineering System Team"
+  $azAdGroupId = Invoke "az ad group show --group 'Azure SDK Engineering System Team' --query id --output tsv"
+
+  Write-Host "Granting 'read/write' access to $appResourceGroupName/$cosmosAccountName"
+  Invoke "az cosmosdb sql role assignment create --resource-group '$appResourceGroupName' --account-name '$cosmosAccountName' --scope '/' --role-definition-id '00000000-0000-0000-0000-000000000002' --principal-id '$azAdGroupId' --output none"
+    
+  if ($LASTEXITCODE -ne 0) {
+    Write-Output $output
+    Write-Error "Failed to grant access"
+    exit 1
+  }
+
+  Write-Host "Granting 'Storage Blob Data Contributor' access to $logsResourceGroupName/$logsStorageAccountName"
+  $scope = "/subscriptions/$subscriptionId/resourceGroups/$logsResourceGroupName/providers/Microsoft.Storage/storageAccounts/$logsStorageAccountName"
+  $output = Invoke "az role assignment create --assignee '$azAdGroupId' --role 'Storage Blob Data Contributor' --scope '$scope' --output none"
+    
+  if ($LASTEXITCODE -ne 0) {
+    Write-Output $output
+    Write-Error "Failed to grant access"
+    exit 1
+  }
+    
+  Write-Host "Granting 'Storage Queue Data Contributor' access to $appResourceGroupName/$appStorageAccountName"
+  $scope = "/subscriptions/$subscriptionId/resourceGroups/$appResourceGroupName/providers/Microsoft.Storage/storageAccounts/$appStorageAccountName"
+  $output = Invoke "az role assignment create --assignee '$azAdGroupId' --role 'Storage Queue Data Contributor' --scope '$scope' --output none"
+    
+  if ($LASTEXITCODE -ne 0) {
+    Write-Output $output
```

Comment:
I'm ok with this but is there a reason you didn't put the role assignments into the bicep templates?

Reply:
The storage assignments are only needed when a dev is running the service locally and connecting to live resources. This doesn't need to be persistent access

Reply:
Also, in the pipeline, the deployment principal doesn't have permission to query Entra.  We'd need to hard code the group's object id into the parameters and that didn't seem correct.